app/features/receive/lightning-address-service.tsx

```diff
@@ -133,21 +138,32 @@ export class LightningAddressService {
         this.cashuReceiveQuoteRepository,
       );
 
-      // We only support BTC for lightning address because we need to get invoices for the exact satoshi amount.
-      // Other currency accounts would require an exchange rate which will create a mismatch in amounts.
+      // For external lightning address requests, we only support BTC to avoid exchange rate mismatches.
+      // However, if the request specifies we can bypass amount validation, we can use the user's default currency
+      // and perform exchange rate conversion to create an invoice in their preferred currency.
       const account = await this.userRepository.getDefaultAccount(
         userId,
-        'BTC',
+        this.request.headers.get('X-Bypass-Amount-Validation') === 'true'
```

Comment:
I would pass this setting through constructor so that this service doesn't have to deal with request and headers. It can be agnostic about how we are propagating this information